src/unix/linux_like/mod.rs

```diff
@@ -1886,6 +1886,14 @@ cfg_if! {
     }
 }
 
+// The statx syscall, available on some libcs.
+cfg_if! {
+    if #[cfg(any(target_env = "gnu", target_os = "android"))] {
+        mod linux_statx;
+        pub use self::linux_statx::*;
```

Comment:
This looks fine to me, but could you inline the definitions here rather than creating an API-specific module?

Reply:
I tried, but I wasn't able to use `s!` and apply `cfg!` conditions

Reply:
Hm, what exactly is the error? This should work okay, see e.g. https://github.com/rust-lang/libc/blob/c3bc4068ad51d91cfe973e62a51169dbb96bce9c/src/unix/linux_like/linux/mod.rs#L1041-L1043

Reply:
@tgross35 not sure what I previously did wrong, but it _seems_ to work now. I'll update the PR, thanks!

Reply:
Ah, the [problems are back](https://github.com/rust-lang/libc/actions/runs/11826922892/job/32953981082?pr=3978), the style check isn't happy:

```
src/unix/linux_like/mod.rs:1892: constant found after extern function when it belongs before
src/unix/linux_like/mod.rs:1933: struct found after constant when it belongs before
src/unix/linux_like/mod.rs:1933: multiple s! macros in one module
```

Reply:
I wound up disabling that `multiple s!` check because it's kind of false-positive-y, so this should be good with a rebase after https://github.com/rust-lang/libc/pull/4107 lands (~1 hour)

Sorry for all the conflicts, been doing a lot of branch cleanup work.

Reply:
However, the other two ordering errors still occur. Perhaps I could move the constants/structs to the other constants/structs

Reply:
It is required that things are ordered typedefs->types->functions->consts, it looks like your `cfg_if` block is reversed.

It it still complains after you correct the ordering, I think you might just need to split into three invocations of `cfg_if!` and sort it with the rest of the file.